src/pudl/constants.py

```diff
@@ -1395,7 +533,7 @@
 """
 
 # The full set of years we currently expect to be able to ingest, per source:
-working_partitions = {
+working_partitions: Dict[str, Any] = {
```

Comment:
Type could be `Dict[str, Dict[str, Union[tuple, str]]]`, or if converted to lists, `Dict[str, Dict[str, Union[List[int], List[str], str]]]`.

Reply:
Oh right, Union.

I've been using Tuples for a bunch of these settings because they're immutable and used as default arguments in a bunch of functions, and we don't want someone accidentally modifying the value in a function and have that change cascade throughout the whole system. Is there a good reason to use List instead?

Reply:
I find tuples awkward for variable-length sequence data entry compared to lists because of the common `(x)` mistake, but that will be caught in the future once static type checking is deployed. The constants are full of mutable things, though, like dictionaries and lists and pandas dataframes.

Anyway, tuples are fine of course :) I've updated the type definitions as needed, using `Tuple[{type}, ...]` for variable-length tuples of a single type.